app/src/main/java/com/example/quizacademytask/AppSingleton.kt

```diff
@@ -0,0 +1,18 @@
+package com.example.quizacademytask
+
+import android.app.Application
+import db.AppDatabase
+
+class AppSingleton : Application() {
+
+    override fun onCreate() {
+        super.onCreate()
+        instance = this
+        db = AppDatabase.getInstance(this)
+    }
+
+    companion object {
+        lateinit var instance: AppSingleton
```

Comment:
you don't need to hold reference to instance of an an app singleton. In rare cases you would need it (and you won't) you could cast application context to your app class. 
It would be better to rename AppSingleton -> App
then App.db would be much better readable than AppSingleton.db